python/ray/train/_internal/session.py

```diff
@@ -48,8 +59,10 @@ def __init__(
         world_rank: int,
         local_rank: int,
         world_size: int,
+        # TODO(xwjiang): Legacy Ray Train trainer clean up!
+        trial_info: Optional[TrialInfo] = None,
         dataset_shard: Optional[Union[Dataset, DatasetPipeline]] = None,
-        checkpoint: Optional[Dict] = None,
+        checkpoint: Optional[Union[Dict, Checkpoint]] = None,
```

Comment:
Any way to simplify this to one type?

Reply:
This will be done after Antoni's work removing legacy ray train work. Added TODO there.